_analyzers/token-filters/synonym-graph.md

```diff
@@ -0,0 +1,180 @@
+---
+layout: default
+title: Synonym graph
+parent: Token filters
+nav_order: 420
+---
+
+# Synonym graph token filter
+
+The `synonym_graph` token filter is a more advanced version of the `synonym` token filter. It supports multi-word synonyms and processes synonyms across multiple tokens, making it ideal for phrases or scenarios in which relationships between tokens are important.
+
+## Parameters
+
+The `synonym_graph` token filter can be configured with the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`synonyms` | Either `synonyms` or `synonyms_path` must be specified | String | A list of synonym rules defined directly in the configuration.
```

Comment:
Nit: Here and on the following line, because they are technically complete sentences, the "Either" sentences should end with a period (I believe, however, that we may not do this on the other pages).